sympy/solvers/solvers.py

```diff
@@ -2742,7 +2742,7 @@ def equal(expr1, expr2):
         elif lhs.is_Mul and rhs.is_positive:
             llhs = expand_log(log(lhs))
             if llhs.is_Add:
-                return _solve(llhs - log(rhs), sym, **flags)
```

Comment:
this looks like a recursion hazard...why is the original form being sent back to solve instead of the log form?

Reply:
The problem was that it was converting the equations here to logarithm form, making information lost for us about even powers.
I would refer to see `_tsolve` code in solve in which it is checking for f.is_Add, f.is_Pow and this one f.is_Mul and after all that it is written in comment that `maybe it is a lambert type`.
https://github.com/sympy/sympy/blob/cc6d08f8e286f83a3f0243a2062392b9825528af/sympy/solvers/solvers.py#L2762-L2765

So what happened was it was converting to logarithm form before coming to _solve_lambert changing the original equation.

I was looking to add here a check like `_is_lambert` or `can_be_lambert` to separate out the bivariate cases.
Or I did some testing it come out that this case doesn't have a single test(all tests passing after removing this case in this PR). May be equations like this are getting solved already before coming to this or some proper test case for this are not added yet.

Reply:
### What equations this case has?
`lhs.is_Mul and rhs.is_positive`

Which means lhs can be Mul of(means like linear*exp where linear and exp are function ox in respective forms) linear, exp, log, trigs.

* So the cases where lhs is like trigs*(linear or exp or log functions of x) are very hard to implement in computer algebraic system and will raise NotImplementedError.

So the cases arise of linear, exp and log
* the cases like linear*exp and linear*log which are of lambert type and are handle already.
* the cases like exp*log are left which can be helpful in this case.
But I tried to run this test
```python
>>> solve(exp(x)*log(x) - 1)
NotImplementedError
```
I don't know much about these type of equations but it is clear that these are very hard equations to implement. Even [wolfram](https://www.wolframalpha.com/input/?i=solve+exp(x)log(x)+-+1+for+x) give the approximated root of this equation .
But our nsolve is raising TypeError to approximate the solution(I don't know if I am expecting right thing from nsolve please correct me if i am wrong).
```python
>>> nsolve(exp(x)*log(x) - 1)
TypeError: cannot create mpf from x
```
So I would say or suggest to remove this case as the equations left out are only bivariate type. @smichr Please let me know what you think of it?

Reply:
I suspect you are right and this case can be removed.